shared/connection.go

```diff
@@ -76,13 +79,15 @@ func MongoSession(opts MongoSessionOpts) *mgo.Session {
 		return nil
 	}
 	session.SetMode(mgo.Eventual, true)
+	session.SetPoolLimit(opts.PoolLimit)
```

Comment:
@timvaillancourt 

Just a side note: https://github.com/go-mgo/mgo/issues/246#issuecomment-199907660

Also you could set pool limit in DSN with `maxPoolSize`:
https://github.com/go-mgo/mgo/blob/v2/session.go#L292-L296

While this certainly makes some sense, I'm not sure about performance gain? @AlekSi was this somehow verified? Would be nice to understand what was the real issue here, if it was not sharing session or lack of this pool limit (which seems weird, because the code is really linear so one request should use one socket).